load-gen/src/main/java/io/smartcat/data/loader/worker/TimeBasedTriggeringWorker.java

```diff
@@ -0,0 +1,66 @@
+package io.smartcat.data.loader.worker;
+
+import java.util.function.Consumer;
+
+import io.smartcat.data.loader.api.Worker;
+
+/**
+* Worker which triggers specified trigger every <code>timeInNanos</code> nanoseconds.
+* It uses time difference to trigger calls. Can not trigger if worker is not called,
+* but if triggered, it is more precise than {@link TimerBasedTriggeringWorker}.
+* Useful for logging or monitoring.
+*
+* @param <T> Type of data this worker accepts.
+*
+* @see TimerBasedTriggeringWorker
+*/
+public class TimeBasedTriggeringWorker<T> implements Worker<T> {
+
+    private final Worker<T> delegate;
+    private final long timeInNanos;
+    private final Consumer<Long> trigger;
+
+    private long last = 0;
+    private long count = 0;
+
+    /**
+     * Constructs worker with specified <code>delegate</code> worker, <code>timeInNanos</code> and
+     * <code>trigger</code>.
+     *
+     * @param delegate Worker which call will be delegated to, cannot be null.
+     * @param timeInNanos Number of nanoseconds after which trigger will be triggered, must be positive number.
+     * @param trigger Consumer which will be triggered every <code>numberOfCalls</code> calls with number of calls as an
```

Comment:
Consider renaming `trigger` into something more appropriate (e.g. consumer). Consumer is not a trigger, it's being triggered.